core/clustersmngr/clustersmngr.go

```diff
@@ -92,20 +98,27 @@ func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfig {
 			}
 		}
 
+		// flowcontrol.IsEnabled makes a request to the K8s API of the cluster stored in the config.
+		// It does a HEAD request to /livez/ping which uses the config.Dial timeout. We can use this
+		// function to error early rather than wait to call client.New.
 		enabled, err := flowcontrol.IsEnabled(context.Background(), config)
-		if err == nil && enabled {
+		if err != nil {
+			return nil, err
```

Comment:
Maybe wrap w/ a message about flow control?

I know that might make the error in the UI a bit wild but I think swinging back in the direction of overly verbose error reporting wouldn't be the worst.